js/packages/web/src/routes.tsx

```diff
@@ -13,27 +13,69 @@ import {
 import { AdminView } from './views/admin';
 import { BillingView } from './views/auction/billing';
 
+import { useLocation } from 'react-router';
+import { useEffect } from 'react';
+import { analyticsInitialized, analyticsUserId, initializeAnalytics, network, pageview, setAnalyticsUserId, setNetwork } from './utils/analytics';
+import { useWallet } from '@solana/wallet-adapter-react';
+
 interface RoutesProps {
   storefront: Storefront;
 }
 
+const Analytics = ({ storefront }: RoutesProps) => {
+  const { publicKey, connected } = useWallet();
+  const pubkey = publicKey?.toBase58() || '';
+  const {endpoint} =useConnectionConfig();
+  const endpointName = ENDPOINTS.find(e => e.endpoint === endpoint)?.name
+
```

Comment:
Thoughts on setting this up as a provider? 

Reference providers.tsx they are all nested within the hashrouter so should have accces to location. And also note to check for a callback handler on route change to track page views.